Yosemite/YosemiteTests/Stores/AppSettingsStoreTests+OrdersSettings.swift

```diff
@@ -0,0 +1,132 @@
+            XCTAssertNil(error)
+        }
+        subject.onAction(writeAction)
+
+
+        // Then
+        let result1: Result<StoredOrderSettings.Setting, Error> = waitFor { promise in
+            let readAction =  AppSettingsAction.loadOrdersSettings(siteID: siteID) { result in
+                promise(result)
+            }
+            self.subject.onAction(readAction)
+        }
+
+        XCTAssertTrue(result1.isSuccess)
+        XCTAssertEqual(try result1.get(), orderSettings)
+    }
+
+    func test_ordersSettings_actions_returns_values_after_being_set_with_two_sites() throws {
+        // Given
+        let siteID1: Int64 = 134
+        let siteID2: Int64 = 268
+
+        let orderStatuses: [OrderStatusEnum] = [.pending, .completed]
+        let startDate = Date().yearEnd
+        let endDate = Date().yearStart
+        let dateRange = OrderDateRangeFilter(filter: .custom, startDate: startDate, endDate: endDate)
+
+        let orderSettings1 = StoredOrderSettings.Setting(siteID: siteID1,
+                                                        orderStatusesFilter: orderStatuses,
+                                                        dateRangeFilter: dateRange)
+        let orderSettings2 = StoredOrderSettings.Setting(siteID: siteID2,
+                                                        orderStatusesFilter: orderStatuses,
+                                                        dateRangeFilter: dateRange)
+
+        // When
+        let writeAction1 = AppSettingsAction.upsertOrdersSettings(siteID: siteID1, orderStatusesFilter: orderStatuses, dateRangeFilter: dateRange) { error in
+            XCTAssertNil(error)
+        }
+        subject.onAction(writeAction1)
+
+        let writeAction2 = AppSettingsAction.upsertOrdersSettings(siteID: siteID2, orderStatusesFilter: orderStatuses, dateRangeFilter: dateRange) { error in
+            XCTAssertNil(error)
+        }
+        subject.onAction(writeAction2)
+
+        // Then
+        let result1: Result<StoredOrderSettings.Setting, Error> = waitFor { promise in
+            let readAction =  AppSettingsAction.loadOrdersSettings(siteID: siteID1) { result in
+                promise(result)
+            }
+            self.subject.onAction(readAction)
+        }
+        XCTAssertTrue(result1.isSuccess)
+        XCTAssertEqual(try result1.get(), orderSettings1)
+
+        let result2: Result<StoredOrderSettings.Setting, Error> = waitFor { promise in
+            let readAction =  AppSettingsAction.loadOrdersSettings(siteID: siteID2) { result in
+                promise(result)
+            }
+            self.subject.onAction(readAction)
+        }
+        XCTAssertTrue(result2.isSuccess)
+        XCTAssertEqual(try result2.get(), orderSettings2)
+    }
+
+    func test_reset_OrdersSettings_action() {
+        let action = AppSettingsAction.resetOrdersSettings
+        subject.onAction(action)
+        XCTAssertTrue(fileStorage!.deleteIsHit)
```

Comment:
nit: would be nice if the tests won't crash when `fileStorage` is nil for some reason

```suggestion
        // When
        let action = AppSettingsAction.resetOrdersSettings
        subject.onAction(action)
        
        // Then
        XCTAssertTrue(fileStorage.deleteIsHit)
```